dashboard/lib/services/lti.rb

```diff
@@ -63,6 +65,31 @@ def self.create_lti_deployment(integration_id, deployment_id)
     )
   end
 
+  def self.generate_auth_id(id_token)
```

Comment:
It should be one operation per [Service Object](https://blog.trk.in.rs/2018/11/30/object-oriented-design-principles-and-patterns/#service-objects):
```ruby
module Services
  module Lti
    class AuthIdGenerator
      def initialize(id_token)
        @id_token = id_token
      end

      def call
        "#{iss}|#{aud}|#{sub}"
      end

      private

      attr_reader :id_token

      def iss
        id_token[:iss]
      end

      def aud
        ...
      end

      def sub
        id_token[:sub]
      end
    end
  end
end
```